src/server/repositories/UrlRepository.ts

```diff
@@ -51,32 +47,40 @@ export class UrlRepository implements UrlRepositoryInterface {
   public findByShortUrl: (
     shortUrl: string,
   ) => Promise<StorableUrl | null> = async (shortUrl) => {
-    return (await Url.findOne({
+    const url = await Url.scope('getClicks').findOne({
       where: { shortUrl },
-    })) as StorableUrl | null
+    })
+    return this.urlMapper.persistenceToDto(url)
   }
```

Comment:
this was an incredibly bad practice to blindly cast the output of `findOne` immediately to the DTO's `StorableUrl` type. it turned off typescript's typechecks and could have been fatal for the server if not caught in development.

I'm adding an extra step of getting the `urlMapper` to convert from raw `Url` to `StorableUrl`. Because this throws when data is incomplete, we can be better assured of type safety.

Reply:
do u know why the previous developer felt it was necessary in the first place?

Reply:
my guess is that previous developer was certain that running `Url.findOne` would always yield an object with all the necessary fields of `StorableUrl`, so did a quick typecast so that the return type matches what's defined in the interface.

But this isn't true anymore, considering a `Url` object by itself is now missing the `clicks` property.

Reply:
i see